### ISSUE TITLE
Code generation causes InconsistentAnalysisException with multiple targets

### ISSUE THREAD

**Issue by eernstg** [google/reflectable.dart#198]
The following commands show that reflectable currently fails to generate code when dependencies have been fully updated (`pub upgrade`):

```
> git clone 'https://github.com/dart-lang/test_reflectable.git'
> cd test_reflectable
> cd tool
> make
```

This causes a number of failures like this:

```
[SEVERE] reflectable:reflectable on test/invoke_test.dart.vm_test.dart:

InconsistentAnalysisException: Requested result might be inconsistent with previously returned results
```

However, it only seems to affect code generation in the case where there are multiple targets, that is, when code generation occurs for multiple separate entry points ("programs").

We do not yet know why this happens, but it may be caused by some changes in the way the build related packages give notifications to the analyzer about changed files, cf. https://github.com/dart-lang/build/issues/2634.

**Comment by eernstg** [comment]
[This PR](https://github.com/google/reflectable.dart/pull/220) uses the workaround from dart-lang/build#2634 to eliminate the inconsistency.

**Comment by eernstg** [comment]
Finally closing this one: Resolved by #220!
